plutus-tx/src/PlutusTx/Coverage.hs

```diff
@@ -86,21 +91,24 @@ data CoverageAnnotation = CoverLocation CovLoc
                         | CoverBool CovLoc Bool
                         deriving (Ord, Eq, Show, Read, Generic, Serialise)
                         deriving Flat via (AsSerialize CoverageAnnotation)
+                        deriving anyclass (NFData, ToJSON, FromJSON, ToJSONKey, FromJSONKey)
 
 instance Pretty CoverageAnnotation where
   pretty (CoverLocation loc) = pretty loc
   pretty (CoverBool loc b)   = pretty loc <+> "=" <+> pretty b
 
 data Metadata = ApplicationHeadSymbol String
+              | IgnoredAnnotation   -- ^ Location that is not interesting to cover.
```

Comment:
I feel like I'm missing something - shouldn't you be introducing these somewhere?

Reply:
Yes, in plutus-apps.

Reply:
I'm still confused. These annotations are generated by the plugin, how are you adding them later?

Maybe you could include some documentation explaining how they're supposed to be used?

Reply:
Some context here: we are about to merge a static analysis tool into plutus-apps that adds these annotations post-fact.